tests/unit/modules/test_nsxt_edge_cluster.py

```diff
@@ -0,0 +1,320 @@
+            "members": [],
+            "cluster_profile_bindings": [
+                {
+                    "resource_type": "EdgeHighAvailabilityProfile",
+                    "profile_id": "91bcaa06-47a1-11e4-8316-17ffc770799b",
+                }
+            ],
+            "member_node_type": "UNKNOWN",
+            "allocation_rules": [],
+            "enable_inter_site_forwarding": False,
+            "resource_type": "EdgeCluster",
+            "id": "1a31ff14-b95d-40b7-882d-e515083ad78b",
+            "display_name": "Sample-Cluster",
+            "description": "",
+            "tags": [],
+            "_create_user": "admin",
+            "_create_time": 1617005898446,
+            "_last_modified_user": "admin",
+            "_last_modified_time": 1617005898446,
+            "_system_owned": False,
+            "_protection": "NOT_PROTECTED",
+            "_revision": 0,
+        },
+    ],
+    "result_count": 2,
+}
+
+_mock_edge_clusters_by_display_name = {
+    "results": [
+        {
+            "deployment_type": "UNKNOWN",
+            "members": [],
+            "cluster_profile_bindings": [
+                {
+                    "resource_type": "EdgeHighAvailabilityProfile",
+                    "profile_id": "91bcaa06-47a1-11e4-8316-17ffc770799b",
+                }
+            ],
+            "member_node_type": "UNKNOWN",
+            "allocation_rules": [],
+            "enable_inter_site_forwarding": False,
+            "resource_type": "EdgeCluster",
+            "id": "5a59778c-47d4-4444-a529-3f0652f3d422",
+            "display_name": "Edge_cluster_101",
+            "description": "",
+            "tags": [],
+            "_create_user": "admin",
+            "_create_time": 1617088848460,
+            "_last_modified_user": "admin",
+            "_last_modified_time": 1617088848460,
+            "_system_owned": False,
+            "_protection": "NOT_PROTECTED",
+            "_revision": 0,
+        }
+    ],
+}
+
+_mock_edge_clusters_create = {
+    "deployment_type": "UNKNOWN",
+    "members": [],
+    "cluster_profile_bindings": [
+        {
+            "resource_type": "EdgeHighAvailabilityProfile",
+            "profile_id": "91bcaa06-47a1-11e4-8316-17ffc770799b",
+        }
+    ],
+    "member_node_type": "UNKNOWN",
+    "allocation_rules": [],
+    "enable_inter_site_forwarding": False,
+    "resource_type": "EdgeCluster",
+    "id": "c4d6ac7f-d331-4929-9532-15e1ddcc77f0",
+    "display_name": "Edge_cluster_Create_Postman",
+    "_create_user": "admin",
+    "_create_time": 1617702887446,
+    "_last_modified_user": "admin",
+    "_last_modified_time": 1617702887446,
+    "_system_owned": False,
+    "_protection": "NOT_PROTECTED",
+    "_revision": 0,
+}
+
+error_json = {"error": "The credentials were incorrect or the account specified has been locked."}
+
+
+@patch.object(nsxt_request, "call_api")
+def test_get_using_basic_auth(mock_call_api):
+
+    mock_call_api.return_value = _mock_edge_clusters
```

Comment:
**suggestion** these should be converted to fixtures. At a glance it looks like there should be two fixtures:

```
@pytest.fixture
def mock_success_api_call(mock_call_api):
    mock_call_api.return_value = {...}  # _mock_edge_clusters
    yield mock_call_api

    
@pytest.fixture
def mock_failure_api_call(mock_call_api):
    mock_call_api.return_value = {...}  # error_json
    yield mock_call_api    
```